third_party/mlir/BUILD

```diff
@@ -1707,7 +1707,7 @@ cc_library(
 
 # TODO(jpienaar): Update this.
 cc_library(
-    name = "MlirOptLib",
+    name = "MlirOptMain",
```

Comment:
This one is in the lib directory while the other one is the main entry point, fix the cmake file instead.

Reply:
Thanks four your feedback. Actually, no of the libraries in `mlir/lib` directory have a `Lib` suffix, so I assume `Main` was chosen as it is the *main* library. In `tools/mlir-opt/` the target is also library, but not the main one and it is linked into the `mlir-opt` binary. I think that is the reason for `Lib` suffix in that case. If you still propose to fix the CMake files, I am glad to do so. However, it seemed that @GMNGeoffrey would prefer this solution as discussed within the IREE pull requests. Anyway, I will be happy to fix this wherever :)

Reply:
I'm fine with either fix, I just assumed that MlirOptMain should refer to the file MlirOptMain. If it's the other way around I suggest the files themselves should also be renamed.